controller/konnect/ops/ops_adopt.go

```diff
@@ -0,0 +1,669 @@
+package ops
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"slices"
+	"sort"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	commonv1alpha1 "github.com/kong/kong-operator/api/common/v1alpha1"
+	kcfgkonnect "github.com/kong/kong-operator/api/konnect"
+	konnectv1alpha1 "github.com/kong/kong-operator/api/konnect/v1alpha1"
+	"github.com/kong/kong-operator/controller/konnect/constraints"
+	sdkops "github.com/kong/kong-operator/controller/konnect/ops/sdk"
+)
+
+// AdoptMatch performs adoption of a Konnect entity in "match" mode. When successful, the existing
+// entity in Konnect is managed by the operator without modifying it.
+func AdoptMatch[
```

Comment:
I have added `Adopt` function in `ops.go` in #2424. It already contains the process of setting status and dispatching to dedicated types. I think it could be better to modify it for the requirement of your implementation.